test/test_image.py

```diff
@@ -33,14 +34,47 @@ def get_images(directory, img_ext):
                 yield os.path.join(root, fl)
 
 
+def pil_read_image(img_path):
+    with Image.open(img_path) as img:
+        return torch.from_numpy(np.array(img))
+
+
+def normalize_dimensions(img_pil):
+    if len(img_pil.shape) == 3:
+        img_pil = img_pil.permute(2, 0, 1)
+    else:
+        img_pil = img_pil.unsqueeze(0)
+    return img_pil
+
+
 class ImageTester(unittest.TestCase):
     def test_decode_jpeg(self):
+        conversion = [(None, 0), ("L", 1), ("RGB", 3)]
         for img_path in get_images(IMAGE_ROOT, ".jpg"):
-            img_pil = torch.load(img_path.replace('jpg', 'pth'))
-            img_pil = img_pil.permute(2, 0, 1)
-            data = read_file(img_path)
-            img_ljpeg = decode_jpeg(data)
-            self.assertTrue(img_ljpeg.equal(img_pil))
+            for pil_mode, components in conversion:
+                with Image.open(img_path) as img:
+                    is_cmyk = img.mode == "CMYK"
+                    if pil_mode is not None:
+                        if is_cmyk:
+                            # libjpeg does not support the conversion
+                            continue
+                        img = img.convert(pil_mode)
+                    img_pil = torch.from_numpy(np.array(img))
+                    if is_cmyk:
+                        # flip the colors to match libjpeg
+                        img_pil = 255 - img_pil
```

Comment:
It seems that this is necessary when comparing CMYK images loaded with PIL vs libjpeg.

Reply:
Hum, this is interesting / weird. Do you have an understanding of why this is the case? And would you say this is a problem with PIL?

Reply:
I do not have a good understanding of why this happens unfortunately. I just noticed that they are represented differently...

Reply:
I would feel a bit more comfortable if I could understand what is going on here. Is OpenCV giving similar results as PIL for the same case if you pass the appropriate reading flags?

Reply:
It seems that this investigation is more meaty that I expected, so I raised a separate issue #3012. Note that what you flagged is an existing behaviour and this PR just adds a test that highlights the issue. Hence I propose not to block the PR for this investigation.

Reply:
Sounds good